src/graccarchive/graccarchive.py

```diff
@@ -235,6 +219,13 @@ def exit_gracefully(signum, frame):
     # Create and run the OverMind
     print "Starting the archiver agent."
     archiver_agent = ArchiverAgent(config)
+    # Capture and call sys.exit for SIGTERM commands
+    def exit_gracefully(signum, frame):
+        archiver_agent.flushFile()
+        archiver_agent.closeFile()
+        sys.exit(0)
+    signal.signal(signal.SIGTERM, exit_gracefully)
```

Comment:
Would it be better to use the atexit module?  That may cover a wider range of code paths.